_posts/2018-11-20-keeping-dependencies-updated.markdown

```diff
@@ -0,0 +1,138 @@
+which is consumed by [Reaction][reaction] (our react component/app library), [Emission][emission] (our React Native
+application), [Force][force] (our main site), and [Positron][positron] (our editorial creation tool). That's not
+even an exhaustive list. As you can imagine, after making an update to [Palette][palette] we have to make a lot of
+Pull Requests to get everything synced up across our many projects. And that's just _one_ dependency.
+
+### Evaluating the problem
+
+There are a few services out there that connect to [GitHub][github] and helps you keep your dependencies up to
+date. I'd personally used [Greenkeeper][greenkeeper] in the past and it seemed to work fairly well for my uses. I'd
+also heard about [Renovate][renovate] which is another option that actually supports more package managers than
+just yarn/npm. Great! Plenty to evaluate here. Anytime I'm evaluating a new service there are a few questions I ask
+myself upfront to help a good decision.
+
+1. What are my exact needs
+2. Can this solution scale to meet future needs
+
+The first point straight-forward, but there's a little twist. We have a _lot_ of dependencies. If we got PRs for
+all of them we'd be pretty much unable to do anything. In this case we wanted to specifically limit it to packages
+that are published by Artsy (on the `@artsy` npm namespace).
+
+The second you have to be a bit careful with. Don't try to project too far or you'll end up choosing a solution far
+too complex for your current needs. In this case, I wanted something that we could selectively extend in the future
+to cover other dependencies. Things like `react` and `react-dom` or `typescript`. Incremental increases without a
+ton of noise.
+
+### Picking a solution
+
+First things first... we have to have a solution that can update only Artsy's dependencies. I started digging
+through [Greenkeeper][greenkeeper]'s docs and found a reference to an [ignore][greenkeeper-ignore] option.
+Essentially any package that you don't want [GreenKeeper][greenkeeper] to automatically update you can put in this
+ignore list. That's not really doable in our usecase because we want to ignore everything but a small subset of
+packages.
+
+Checking out [Renovate][renovate]'s docs I found a more promising option:
+[excludePackagePatterns][renovate-exclude]. All I really want to do is include Artsy packages, but this sounded
+like I could do the inverse by excluding all non-Artsy packages. Being as it had that option, supported more
+package managers, and had a more friendly pricing scheme than [Greenkeeper][greenkeeper] I decided to give
+[Renovate][renovate] a shot.
+
+### Making it happen
+
+Started by enabling [Renovate][renovate] on [Force][force]. You can see the PR [here][renovate-pr].
+[Renovate][renovate] has a really excellent on-boarding experience where it creates an initial PR with the
+configuration that'll let you know which packages will be updated before it ever creates a PR to update anything.
```

Comment:
This is the one sentence I found a little difficult to parse. Would suggest more bitesized chunks for low-attention-span infinite-scrollers like me.

e.g. Renovate has a really excellent on-boarding experience. The first thing it does is submit a PR to configure itself. This lets you see and decide which packages it will keep up-to-date.

Reply:
Okay, I played around with it a little bit. Let me know if the new wording flows better.